torch_xla_py/xla_dist_test.py

```diff
@@ -3,15 +3,113 @@
 from __future__ import print_function
 
 import unittest
-import xla_dist
+from xla_dist import ClientWorker
+from xla_dist import ServiceWorker
+from xla_dist import Cluster
+
 
 class ClusterTest(unittest.TestCase):
 
-  def test_validate_service_cluster(self):
-    pass
+  def test_validate_good_cluster(self):
+    client_workers = [
+        ClientWorker('10.0.0.0', 'n1-standard-16', 'europe-west4-a'),
+        ClientWorker('10.0.0.1', 'n1-standard-16', 'europe-west4-a'),
+        ClientWorker('10.0.0.2', 'n1-standard-16', 'europe-west4-a'),
+        ClientWorker('10.0.0.3', 'n1-standard-16', 'europe-west4-a',
+                     hostname='test'),
+    ]
+    service_workers = [
+        ServiceWorker('10.0.0.0', 'v3-32', 'europe-west4-a'),
+        ServiceWorker('10.0.0.1', 'v3-32', 'europe-west4-a'),
+        ServiceWorker('10.0.0.2', 'v3-32', 'europe-west4-a'),
+        ServiceWorker('10.0.0.3', 'v3-32', 'europe-west4-a'),
+    ]
+    cluster = Cluster(client_workers, service_workers)
+    cluster.validate()  # Does not raise exception
 
-  def test_validate_client_cluster(self):
-    pass
+  def test_create_bad_client_workers(self):
+    service_workers = [
+        ServiceWorker('10.0.0.1', 'v3-8', 'europe-west4-a'),
+    ]
+    self.assertRaisesRegex(
+        ValueError,
+        'client_workers argument must be a list of ClientWorker',
```

Comment:
should we make these error messages class variables and use those here in tests as opposed to strings that are subject to change?